### ISSUE TITLE
Support validator removal

### ISSUE THREAD

**Issue by ruuda** [ChorusOne/solido#152]
At some point, we will need to support removing validators. I’m opening this issue so I can reference it from TODO comments.

Removal is tricky, because before we can remove a validator, we need to make sure that no stake is delegated to it. While active stake is deactivating, we should also prevent new stake from being delegated to it. I think we need a two-stage process: first we mark the validator for removal, which prevents new delegations, and which should allow the maintainer bots to deactivate stake. Then once all stake has been deactivated, the maintainer bots can remove the validator from the Lido state entirely.

**Comment by ruuda** [comment]
Superseded by #347.
